tequilapi/endpoints/proposals_filter.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package endpoints
+
+import (
+	"github.com/mysteriumnetwork/node/core/discovery/reducer"
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/mysteriumnetwork/node/market/mysterium"
+)
+
+// proposalsFilter defines all flags for proposal filtering in discovery of Mysterium Network
+type proposalsFilter struct {
```

Comment:
This is a weird component. 

I can create it once, but change its state after creation, which will affect the .Matches behaviour. I suggest adding a constructor and making the fields private. That way, I'll know that the behaviour won't change.

Reply:
Made attributes private